lib/option.js

```diff
@@ -11,21 +11,8 @@ class Option {
    */
 
   constructor(flags, description) {
-    this.flags = flags;
-    this.description = description || '';
-
-    this.required = flags.includes('<'); // A value must be supplied when the option is specified.
-    this.optional = flags.includes('['); // A value is optional when the option is specified.
-    // variadic test ignores <value,...> et al which might be used to describe custom splitting of single argument
-    this.variadic = /\w\.\.\.[>\]]$/.test(flags); // The option can take multiple values.
+    this.setFlagsAndDescription(flags, description);
```

Comment:
I was quite surprised this refactor did not break the `typescript-checkJS` run-script. At one time I am pretty sure we were getting errors for JavaScript properties which were not declared/set in the constructor.

TypeScript may have removed that check as too strict for real-world JavaScript.

https://github.com/microsoft/TypeScript/issues/22896